### ISSUE TITLE
About file organization and caching

### ISSUE THREAD

**Issue by pat-s** [krlmlr/vistransrep#20]
I've seen that you make use of Rmd childs quite heavily.

This is nice for file organization. However it is tedious if once wants to add a chapter/section (especially section) later on. All files need to be renamed to maintain the naming scheme.

Another, maybe more practical issue, is the case of caching.
Due our work on the [mlr3book](https://mlr3book.mlr-org.com/), we've found that caching only works reliable on the chapter level.

Meaning if there is one Rmd per chapter (with one l1 heading) and one sets 

```r
knitr::opts_chunk$set(cache = TRUE, cache.lazy = FALSE)
```

Then caching via `bookdown::serve_book()` works reliable.
We've had a similar nested file structure before and it was just a pain.

I don't want to force a change but just raise a discussion about this. Maybe you agree and also feel more comfy with this when adding/changing content.

Downside: Somewhat large Rmd's which contain the content of complete chapters.
At least one can navigate using the RStudio Rmd outline. But at some points its tedious.

---

Another hint: We've also written a small fun for automatic naming of the chunks of each Rmd: https://github.com/mlr-org/mlr3book/blob/03374e976b8d767d7088579bb878bc774d133199/R/name_chunks.R

**Comment by krlmlr** [comment]
`qmv` largely solves the file reorganization issues.

The children are tied to individual scripts which I'd like to maintain. Merging might be possible but will involve infrastructure change. Maybe later?

**Comment by pat-s** [comment]
Ok I see. What is `qmv` and how do I use it? Never heard of it and google didn't help.

**Comment by krlmlr** [comment]
On some platforms you can `apt install renameutils`.

https://www.nongnu.org/renameutils/

**Comment by krlmlr** [comment]
`qmv .` opens an editor where you can edit a rename recipe for all files in the directory. After quitting the recipe is applied.

**Comment by pat-s** [comment]
Looks interesting. Thanks!

**Comment by pat-s** [comment]
I've tried `qmv`. While it helps for bulk-renaming, it does not help when adding a new "01-foo" file to a list of files spanning a integer prefix range from 1-X. I need to touch every file then. Or did I miss something?

**Comment by krlmlr** [comment]
I agree it removes some of the pain, not all of it. Maybe you can teach your text editor to increase numbers in a clever way? Gaps in the sequence help too.

**Comment by krlmlr** [comment]
I prefer caching off for now.
